# Conclusion

In [37]:
print("\n--- Model Performance Comparison (Original Scale) ---")
print(f"Random Forest Regressor: MSE = {mse_rf_original:.4f}, R2 = {r2_rf_original:.4f}")
print(f"XGBoost Regressor:       MSE = {mse_xgb_original:.4f}, R2 = {r2_xgb_original:.4f}")


--- Model Performance Comparison (Original Scale) ---
Random Forest Regressor: MSE = 367.5128, R2 = 0.9917
XGBoost Regressor:       MSE = 857.6206, R2 = 0.9806


## Final Task

### Subtask:
Summarize the process of trying a different machine learning model (XGBoost), its performance, and insights gained from comparing it with the Random Forest model.


## Summary:

### Q&A
The process involved importing, training, and evaluating an XGBoost Regressor model and then comparing its performance against a previously established Random Forest Regressor model. The insights gained from this comparison indicate that while both models performed similarly well on log-transformed data, the Random Forest Regressor demonstrated superior performance on the original scale of the CO2 emissions data.

### Data Analysis Key Findings
*   An XGBoost Regressor model was successfully initialized and trained with `n_estimators=100`, `random_state=42`, and `n_jobs=-1`.
*   On the **log-transformed scale**, the XGBoost model achieved a Mean Squared Error (MSE) of `0.0427` and an R-squared (R2) of `0.9830`.
*   On the **original scale** (after inverse transformation), the XGBoost model yielded an MSE of `857.6206` and an R2 of `0.9806`.
*   **Comparison with Random Forest (Log-Transformed Scale)**:
    *   Random Forest: MSE = `0.0424`, R2 = `0.9831`
    *   XGBoost: MSE = `0.0427`, R2 = `0.9830`
    *   Both models showed comparable and high performance, with Random Forest being marginally better.
*   **Comparison with Random Forest (Original Scale)**:
    *   Random Forest: MSE = `367.5128`, R2 = `0.9917`
    *   XGBoost: MSE = `857.6206`, R2 = `0.9806`
    *   The Random Forest Regressor significantly outperformed XGBoost on the original scale, exhibiting a much lower MSE (less than half of XGBoost's) and a higher R-squared value.

### Insights or Next Steps
*   For this specific problem, the Random Forest Regressor is the preferred model due to its significantly better performance on the original scale of the CO2 emissions, providing more accurate and interpretable predictions.
*   Further hyperparameter tuning for the XGBoost model could be explored to potentially improve its performance on the original scale, as its current results might be influenced by default parameters.
